packages/headless/src/features/query-set/query-set-slice.test.ts

```diff
@@ -101,6 +102,27 @@ describe('querySet slice', () => {
     expect(nextState).toEqual(expectedQuerySet);
   });
 
+  it('sets all queries to q on restoreSearchParameters, when defined', () => {
+    registerEmptyQueryWithId('foo');
+    registerEmptyQueryWithId('bar');
+
+    const expectedQuerySet = {foo: 'world', bar: 'world'};
+    const nextState = querySetReducer(
+      state,
+      restoreSearchParameters({q: 'world'})
+    );
+    expect(nextState).toEqual(expectedQuerySet);
+  });
+
+  it('sets all queries to an empty string on restoreSearchParameters, when not defined', () => {
```

Comment:
When a parameter is absent from the `restoreSearchParameters` payload, the behaviour consistent with the other parameters would be a no-op.

This allows developers to dispatch the action with different values, without needing to specify active values:
```
dispatch(restoreSearchParameters({q: 'hello'}));
dispatch(restoreSearchParameters({sortCriteria: 'relevance'})); // does not reset query to an empty string
```

This is different from the `synchronize` method on the url and search parameter managers, in which case omitting a value means "set to the default value".

Reply:
I see, wasn't sure about this one...